### PR TITLE
Parse json values on getValue

### PR DESCRIPTION
**Pull request by azriel46d**
Hi, this PR is to introduce to `getValue` an options object. 
Rather than returning a `string` or numeric value,  in this way , the developer can pass in an additional options object to indicate that the value is json and a possible fallback solution. The fallback is utilised as part of a try/catch setup just in case there is malformed json.

example 
```javascript
/** The Feature
{
    "feature": {
      "name": "general.languages"
    },
    "feature_state_value": "[\"en-GB\"]",
    "enabled": true
  },
*/

getValue('general.languages', { json: true, fallback:  ["en-AU"] })
```

### CODE REVIEW

File: lib/flagsmith/src/flagsmith-core.ts [hunk 1]
```diff
@@ -622,16 +622,23 @@ const Flagsmith = class {
         this.updateEventStorage();
     }
 
-    getValue = (key:string) => {
+    getValue = (key:string, options?: GetValueOptions) => {
```
Comment (kyle-ssg) [inline lib/flagsmith/src/flagsmith-core.ts:625]: If I'm not mistaken, this means we need to change https://github.com/Flagsmith/flagsmith-js-client/pull/147/files#diff-e2f04af8a0e801f4615f5e4e9d767731aebcb8644569cfdf168d8b21fb202ee7R114 and either IFlagsmithValue or the return type to ``IFlagsmithValue| Array<any> | object``

**Comment by kyle-ssg** [comment]
Great concept, should improve the dev experience for sure ❤️ ! I just have 1 comment on the PR, other than that I'll give it a quick run-through/update docs then I think we can get this published.

**Comment by kyle-ssg** [comment]
With how this is implemented, this could also be used on getTrait() but I think we can leave that for now

File: lib/flagsmith/src/flagsmith-core.ts (hunk 1 above)
Comment (kyle-ssg) [reply lib/flagsmith/src/flagsmith-core.ts:625]: I wonder if we can actually use generics here ``getValue<T= Array<any> | object> `` and ``GetValueOptions<T>``

Comment (azriel46d) [reply lib/flagsmith/src/flagsmith-core.ts:625]: Missed the entry for `getValue` thanks for that :)
Typings should be now resolved thanks to the help of @dangrima90  

https://user-images.githubusercontent.com/2339134/193662430-ac4b2095-4ace-4aef-a449-44d736479b8a.mov

**Comment by kyle-ssg** [comment]
Merged, I'm going to play around with it locally and then publish it later today.

**Comment by kyle-ssg** [comment]
This is released in 3.10.5 https://github.com/Flagsmith/flagsmith-js-client/releases/tag/3.10.5 🚀

**Pull request merged**